### DISCUSSION TITLE
Change accept header in Request class

### DISCUSSION THREAD

**Discussion by sinnbeck** [saloonphp/saloon#439] [Q&A]
On my connector I have set the Accept header. The problem is that I need use another Accept header on one specific request class. Is there a way to overwrite it the class? I have tried setting it both in `defaultHeaders()` and `headers()` but none of them seem to do anything. I have also tried adding the `AcceptsJson` trait, but that does not seem to do anything either

**Comment by sinnbeck** [comment]
For now I have added a method to the connector for overwriting it, but it feels like something I should be able to set on the request :)

```php
    public function acceptJson()
    {
        $this->headers()->add('Accept', 'application/json');

        return $this;
    }

**Reply by juse-less** [reply]
@sinnbeck Hey.
Hmm, that does sound weird.

Are you sure you added it [like this](https://docs.saloon.dev/the-basics/requests#headers)?

If so, are you possibly using a trait on the Connector, that sets the header?
I'm thinking if [this is your issue](https://github.com/saloonphp/saloon/issues/398).
If so, maybe try the [boot method](https://docs.saloon.dev/digging-deeper/middleware#the-boot-method).
